### PR TITLE
[IGNORE][Part of review of PR 147] GH-70 State service amendments 1

### PR DESCRIPTION
**Pull request by lucafavatella**
I started reviewing PR #147, and in the meantime I coded a few fixes for understanding it better. Feel free to merge this - in `gh-70-state-trees-service` branch - and then squash if you want.

Now I will focus more on reviewing interactions between state and chain services.

### CODE REVIEW

File: apps/aecore/src/aec_state.erl [hunk 1]
```diff
@@ -123,10 +120,11 @@ code_change(_OldVsn, State, _Extra) ->
 setup_trees(0, Trees) ->
     Trees;
 setup_trees(N, Trees) ->
+    TreesAtEndOfPreviousBlock = setup_trees(N-1, Trees),
     {ok, Block} = aec_chain:get_block_by_height(N),
     Txs = aec_blocks:txs(Block),
-    {ok, TreesUpdated} = apply_txs_internal(Txs, N, Trees),
-    setup_trees(N-1, TreesUpdated).
```
Comment (sennui) [inline apps/aecore/src/aec_state.erl]: OK, something here is not correct, but we need recursion to initialize state from block at given height.

This should go from genesis to N, not from N to genesis.

Comment (lucafavatella) [reply apps/aecore/src/aec_state.erl]: Recursion is in line 123.

Comment (lucafavatella) [reply apps/aecore/src/aec_state.erl]: It is using stack.

Comment (sennui) [reply apps/aecore/src/aec_state.erl]: ok, sorry I missed that. I will refactor to stack-less solution. It may be very heavy with long chaing

Comment (lucafavatella) [reply apps/aecore/src/aec_state.erl]: ok

Comment (lucafavatella) [reply apps/aecore/src/aec_state.erl]: @sennui Feel free to just close this PR and include/squash some/all these changes in PR 147 (even force-pushing is ok).

Comment (sennui) [reply apps/aecore/src/aec_state.erl]: merging this, fixing and will squash

**Pull request merged**